### PR TITLE
Add array of required properties to inputSchema of Tool

### PR DESCRIPTION
**Pull request by strowk**
Add list of required properties to json schema of Tool under inputSchema.

## Motivation and Context
inputSchema in Tool has "type" and "properties", but does not seem to have "required", while actual implementations appear to be providing this, f.e: https://github.com/modelcontextprotocol/servers/blob/0ff39c72c618976fb06e5a6be902ffd468f368c0/src/sentry/src/mcp_server_sentry/server.py#L241
https://github.com/modelcontextprotocol/servers/blob/0ff39c72c618976fb06e5a6be902ffd468f368c0/src/sqlite/src/mcp_server_sqlite/server.py#L284

## How Has This Been Tested?
I assume since a bunch of python based servers use this, it is tested there already.

## Breaking Changes
There should not be any, as "required" itself is not required.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update

## Checklist
- [x] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [ ] ~~I have added appropriate error handling~~ - not applicable
- [x] I have added or updated documentation as needed

## Additional context
I was trying to make my own server based of existing code, however relying on json schema turned out to be problematic, as it does not always correspond to what actual implementations are doing. This change attempts to fix it.

### CODE REVIEW

File: schema/schema.json [hunk 1]
```diff
@@ -1987,6 +1987,12 @@
                         "type": {
                             "const": "object",
                             "type": "string"
+                        },
+                        "required": {
+                            "items": {
+                                "type": "string"
+                            },
+                            "type": "array"
```
Comment (dsp-ant) [inline schema/schema.json]: Note that schema.json is generated from schema.ts. Please follow [CONTRIBT](https://github.com/modelcontextprotocol/specification/blob/main/CONTRIBUTING.md)

**Review by dsp-ant — CHANGES_REQUESTED** [review]

File: schema/schema.json (hunk 1 above)
Comment (strowk) [reply schema/schema.json]: Got it! [This should do it](https://github.com/modelcontextprotocol/specification/pull/87/commits/d110191ba4ad280ab2282b228975a60f01187235) , thanks

**Review by dsp-ant — APPROVED** [review]
Thank you so much. This is a good catch.

**Review by jspahrsummers — APPROVED** [review]
I think, semantically, this doesn't change anything, since object types are always open-ended in our schema. But if it's useful for clarity, I'm supportive.

**Comment by dsp-ant** [comment]
> I think, semantically, this doesn't change anything, since object types are always open-ended in our schema. But if it's useful for clarity, I'm supportive.

Probably also helps quite a bit if people want to generate language implementations from the json schema.

**Pull request merged**